src/core/cgroup.c

```diff
@@ -37,19 +37,34 @@
 #include "stdio-util.h"
 #include "string-table.h"
 #include "string-util.h"
+#include "virt.h"
 
 #define CGROUP_CPU_QUOTA_PERIOD_USEC ((usec_t) 100 * USEC_PER_MSEC)
 
+bool manager_owns_root_cgroup(Manager *m) {
+        assert(m);
+
+        /* Returns true if we are managing the root cgroup. Note that it isn't sufficient to just check whether the
+         * group root path equals "/" since that will also be the case if CLONE_NEWCGROUP is in the mix. Since there's
+         * appears to be no nice way to detect whether we are in a CLONE_NEWCGROUP namespace we instead just check if
+         * we run in any kind of container virtualization. */
+
+        if (detect_container() > 0)
+                return false;
```

Comment:
Here, and in the other place, I'm not sure if it wouldn't be better to use `!= 0`. I think it's likely that we won't be able to query this properly in some containment scenarios.

Reply:
my thinking was the obverse: if we can't figure things out then assume we are on some weird form of a host... i.e. the thinking is: unless we know for sure we are in a container we are on the host.

But dunno, I figure we could revisit this all over the tree, separately.